### PR TITLE
Fix auto-update channel checking and pinning

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

Fixes error from auto-update `nextUpdateChannel` and from the auto-updater not using the current channel by default.

### CODE REVIEW

File: src/common/user-store/preferences-helpers.ts [hunk 1]
```diff
@@ -272,6 +272,7 @@ const updateChannels = new Map([
     label: "Alpha"
   }],
 ]);
+const defaultUpdateChannel = new SemVer(getAppVersion()).prerelease[0].toString();
```
Comment (jim-docker) [inline src/common/user-store/preferences-helpers.ts]: For dev builds `.prerelease` might not be defined?

Comment (Nokel81) [reply src/common/user-store/preferences-helpers.ts]: Ah good catch, should default to `"latest"` then

**Review by jim-docker — APPROVED** [review]
LGTM, but didn't test, see if it works for next prereleases

**Comment by Nokel81** [comment]
@jim-docker sounds good, it isn't really possible to test any other way as there doesn't seem to be a way to change where the updater looks for updates from.

**Pull request merged**